trace/http.go

```diff
@@ -2,19 +2,46 @@ package trace
 
 import "net/http"
 
-func Get(url string, parent *http.Request) (resp *http.Response, err error) {
-	c := &http.Client{}
-	req, err := http.NewRequest("GET", url, nil)
-	if err != nil {
-		return nil, err
+// DefaultCollector is a MemoryCollector. Trace defaults back to DefaultCollector if none given.
+var DefaultCollector = NewMemoryCollector()
+
+// Trace is used to add incoming Trace functionality.
+type Trace struct {
+	collector Collector
+}
+
+func (t *Trace) process(rw http.ResponseWriter, r *http.Request, f http.HandlerFunc) {
+	if t.collector == nil {
+		t.collector = DefaultCollector
 	}
-	SetIdIfMissing(req, parent)
-	return c.Do(req)
+
+	span := NewSpanIDFromRequest(r)
+	event := NewEvent(span, RequestReceived)
+	t.collector.Record(event)
+
+	r2 := SetHeaders(r, span)
+	f(rw, r2)
+
+	event = NewEvent(span, RequestCompleted)
+	t.collector.Record(event)
+}
+
+// Handler wraps an existing http.Handler with Trace functionality.
+func (t *Trace) Handler(h http.Handler) http.Handler {
```

Comment:
I would change the signature here to Handler be just a simple function. I think generating an `http.Handler` that uses a tracer is not the responsibility of the `Trace` object.

I would do something like:

```go
func Handler(t *Trace, h http.Handler) http.Handler) {
    ....
}
```

And then use it like:

```go
http.ListenAndServe(":8080", trace.Handler(t, mux))
```